packages/core/src/factory.ts

```diff
@@ -14,60 +18,66 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
   defaultConfig: Config,
   components: IPrismComponents<Resource, Input, Output, Config>
 ): IPrism<Resource, Input, Output, Config> {
+  type ResourceAndValidation = {
```

Comment:
just to be sure, this was intentional for the type to be put inside `factory` ?

Reply:
It is, because the `Resource` generic type is defined there. Putting it outside would not help. This is another place we'll need to break, but not in this PR.